src/software/jetson_nano/services/motor.cpp

```diff
@@ -93,9 +125,137 @@ std::unique_ptr<TbotsProto::DriveUnitStatus> MotorService::poll(
+//             ───────────────►       │        │   ┌─────────┐   │
+//                            │       │SEL(HIGH)   │         │   │
+//                            │       ├────────┬───►TMC6100  │   │
+//                            │       │        │   └─────────┘   │
+//                            │       │        │                 │
+//                            └───▲───┘        └─────────────────┘
+//                                │
+//                                │ SEL
+//                                │
+//                  spi_cs_driver_to_controller_demux
+//
+uint8_t MotorService::tmc4671ReadWriteByte(uint8_t motor, uint8_t data,
+                                           uint8_t last_transfer)
+{
+    spi_cs_driver_to_controller_demux_gpio.setValue(GpioState::LOW);
+    return readWriteByte(motor, data, last_transfer);
+}
+
+uint8_t MotorService::tmc6100ReadWriteByte(uint8_t motor, uint8_t data,
+                                           uint8_t last_transfer)
+{
+    spi_cs_driver_to_controller_demux_gpio.setValue(GpioState::HIGH);
+    return readWriteByte(motor, data, last_transfer);
+}
+
+uint8_t MotorService::readWriteByte(uint8_t motor, uint8_t data, uint8_t last_transfer)
+{
+    uint8_t ret_byte = 0;
+
+    if (!transfer_started)
+    {
+        memset(tx, 0, sizeof(tx));
+        memset(rx, 0, sizeof(rx));
+        position = 0;
+
+        if (data & TMC_WRITE_BIT)
+        {
+            // If the transfer started and its a write operation,
+            // set the appropriate flags.
+            currently_reading = false;
+            currently_writing = true;
+        }
+        else
+        {
+            // The first byte should contain the address on a read operation.
+            // Trigger a transfer (1 byte) and buffer the response (4 bytes)
+            tx[position] = data;
+            spiTransfer(file_descriptors[motor], tx, rx, 5);
+
+            currently_reading = true;
+            currently_writing = false;
+        }
+
+        transfer_started = true;
+    }
+
+    if (currently_writing)
+    {
+        // Buffer the data to send out when last_transfer is true.
+        tx[position++] = data;
+    }
+
+    if (currently_reading)
+    {
+        // If we are reading, we just need to return the buffered data
+        // byte by byte.
+        ret_byte = rx[position++];
+    }
+
+    if (currently_writing && last_transfer)
+    {
+        // we have all the bytes for this transfer, lets trigger the transfer and
+        // reset state
+        spiTransfer(file_descriptors[motor], tx, rx, 5);
+        transfer_started = false;
+    }
+
+    if (currently_reading && last_transfer)
+    {
+        // when reading, if last transfer is true, we just need to reset state
+        transfer_started = false;
+    }
+
+    return ret_byte;
+}
 
 void MotorService::start()
 {
+    driver_control_enable_gpio.setValue(GpioState::HIGH);
     // TODO (#2332)
```

Comment:
is this TODO still valid?

Reply:
yeah it is, this is part 1

part 2 is gonna be the rest of it